sktime/forecasting/var.py

```diff
@@ -0,0 +1,123 @@
+__author__ = ["Taiwo Owoseni"]
+
+from statsmodels.tsa.api import VAR as _VAR
+from sktime.forecasting.base.adapters import _StatsModelsAdapter
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+import numpy as np
+
+
+class VectorAutoRegression(_StatsModelsAdapter):
+    """
+    A VAR model is a generalisation of the univariate autoregressive.
+
+    A model for forecasting a vector of time series[1].
+
+    Parameters
+    ----------
+    dates: np.ndarray, optional (default=None)
+        An array-like object of datetime objects
+    freq: str, optional (default=None)
+        The frequency of the time-series.
+        Pandas offset or ‘B’, ‘D’, ‘W’, ‘M’, ‘A’, or ‘Q’.
+    missing: str, optional (default='none')
+        A string specifying if data is missing
+
+    References
+    ----------
+    [1] Athanasopoulos, G., Poskitt, D. S., & Vahid, F. (2012).
+    Two canonical VARMA forms: Scalar component models vis-à-vis the echelon form.
+    Econometric Reviews, 31(1), 60–83, 2012.
+
+    Example
+    -------
+    >>> from sktime.forecasting.var import VectorAutoRegression as VAR
+    >>> from sktime.forecasting.model_selection import temporal_train_test_split
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> import pandas as pd
+    >>> index = pd.date_range(start="2005", end="2006-12", freq="M")
+    >>> df = pd.DataFrame(np.random.randint(0, 100, size=(23, 2)),
+    ... columns=list("AB"),
+    ... index=pd.PeriodIndex(index))
+    >>> train, test = temporal_train_test_split(df)
+    >>> sktime_model = VAR()
+    >>> fh = ForecastingHorizon([1, 3, 4, 5, 7, 9])
+    >>> sktime_model.fit(train)
+    VectorAutoRegression(dates=None, freq=None, missing='none')
+    >>> y_pred = sktime_model.predict(fh=fh)
+    """
+
+    _fitted_param_names = ("aic", "fpe", "hqic", "bic")
+
+    _tags = {
+        "scitype:y": "multivariate",
+        "y_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "univariate-only": False,
+    }
+
+    def __init__(self, dates=None, freq=None, missing="none"):
+        # Model params
+        self.dates = dates
+        self.freq = freq
+        self.missing = missing
+
+        super(VectorAutoRegression, self).__init__()
+
+    def _fit_forecaster(self, y, X=None):
+        """Fit forecaster to training data.
+
+        Wraps Statsmodel's VAR fit method.
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            Target time series to which to fit the forecaster.
+        fh : int, list, np.array or ForecastingHorizon, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        self._forecaster = _VAR(
+            y, dates=self.dates, freq=self.freq, missing=self.missing
+        )
+        self._fitted_forecaster = self._forecaster.fit()
```

Comment:
We need to pass additional parameters here to `fit` (see https://www.statsmodels.org/stable/generated/statsmodels.tsa.vector_ar.var_model.VAR.fit.html#statsmodels.tsa.vector_ar.var_model.VAR.fit), e.g.

```suggestion
        self._fitted_forecaster = self._forecaster.fit(maxlags=self.maxlags, ...)
```
Take a look at sktime's ExponentialSmoothing class where we pass these parameters to the constructor. We then need to pass these values here to the `fit` method of the statsmodels class.

Reply:
Done. nut assigned trend as "c"

Reply:
When trend is constant-c and index of the fh is PeriodIndex, the forecasted time period are all the same. I am thinking of making the default trend not-constant - nc.

Reply:
@mloning

Reply:
We should use the same default as statsmodels here I think

Reply:
Okay. That is "c". I will not change it then